src/distribution/laplace.rs

```diff
@@ -0,0 +1,363 @@
+                * (1.0 - f64::exp(-f64::abs(x - self.location) / self.scale))))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use distribution::internal::*;
+    use distribution::{laplace, Continuous, Univariate};
+    use rand::distributions::Distribution;
+    use rand::thread_rng;
+    use statistics::*;
+    use std::f64;
+
+    fn try_create(location: f64, scale: f64) -> laplace::Laplace {
+        let n = laplace::Laplace::new(location, scale);
+        assert!(n.is_ok());
+        n.unwrap()
+    }
+
+    fn create_case(location: f64, scale: f64) {
+        let n = try_create(location, scale);
+        assert_eq!(location, n.location);
+        assert_eq!(scale, n.scale);
+    }
+
+    fn bad_create_case(location: f64, scale: f64) {
+        let n = laplace::Laplace::new(location, scale);
+        assert!(n.is_err());
+    }
+
+    fn test_case<F>(location: f64, scale: f64, expected: f64, eval: F)
+    where
+        F: Fn(laplace::Laplace) -> f64,
+    {
+        let n = try_create(location, scale);
+        let x = eval(n);
+        assert_eq!(expected, x);
+    }
+
+    fn test_is_nan<F>(location: f64, scale: f64, eval: F)
+    where
+        F: Fn(laplace::Laplace) -> f64,
+    {
+        let n = try_create(location, scale);
+        let x = eval(n);
+        assert!(x.is_nan());
+    }
+
+    fn test_almost<F>(location: f64, scale: f64, expected: f64, acc: f64, eval: F)
+    where
+        F: Fn(laplace::Laplace) -> f64,
+    {
+        let n = try_create(location, scale);
+        let x = eval(n);
+        assert_almost_eq!(expected, x, acc);
+    }
+
+    #[test]
+    fn test_create() {
+        try_create(1.0, 2.0);
+        try_create(f64::NEG_INFINITY, 0.1);
+        try_create(-5.0 - 1.0, 1.0);
+        try_create(0.0, 5.0);
+        try_create(1.0, 7.0);
+        try_create(5.0, 10.0);
+        try_create(f64::INFINITY, f64::INFINITY);
+    }
+
+    #[test]
+    fn test_create_fail() {
+        bad_create_case(2.0, -1.0);
+        bad_create_case(f64::NAN, 1.0);
+        bad_create_case(f64::NAN, -1.0);
+    }
+
+    #[test]
+    fn test_mean() {
+        test_case(f64::NEG_INFINITY, 0.1, f64::NEG_INFINITY, |l| l.mean());
+        test_case(-5.0 - 1.0, 1.0, -6.0, |l| l.mean());
+        test_case(0.0, 5.0, 0.0, |l| l.mean());
+        test_case(1.0, 10.0, 1.0, |l| l.mean());
+        test_case(f64::INFINITY, f64::INFINITY, f64::INFINITY, |l| l.mean());
+    }
+
+    #[test]
+    fn test_variance() {
+        test_almost(f64::NEG_INFINITY, 0.1, 0.02, 1E-12, |l| l.variance());
+        test_almost(-5.0 - 1.0, 1.0, 2.0, 1E-12, |l| l.variance());
+        test_almost(0.0, 5.0, 50.0, 1E-12, |l| l.variance());
+        test_almost(1.0, 7.0, 98.0, 1E-12, |l| l.variance());
+        test_almost(5.0, 10.0, 200.0, 1E-12, |l| l.variance());
+        test_almost(f64::INFINITY, f64::INFINITY, f64::INFINITY, 1E-12, |l| {
+            l.variance()
+        });
+    }
+
+    #[test]
+    fn test_stddev() {
+        test_almost(
```

Comment:
We've been using `#[cgf_attr(rustfmt, rustfmt_skip)]` to prevent the formatter from mangling test code. See [here](https://github.com/boxtown/statrs/blob/master/src/distribution/normal.rs#L307)